lib/resourcemerge/core.go

```diff
@@ -466,12 +466,10 @@ func ensureVolumeSourceDefaults(required *corev1.VolumeSource) {
 }
 
 func ensureSecurityContextPtr(modified *bool, existing **corev1.SecurityContext, required *corev1.SecurityContext) {
-	// if we have no required, then we don't care what someone else has set
-	if required == nil {
+	if *existing == nil && required == nil {
 		return
 	}
-
-	if *existing == nil {
+	if *existing == nil || (required == nil && *existing != nil) {
```

Comment:
How does it handle `existing != nil , required != nil  but existing != require ` case? 
Can we add code comments about which cases the conditional loops are handling?

Reply:
If both are nil, simply return.
If existing is not set, then set existing to required.
Is existing is set and required is not set, then set existing to required.
Otherwise, go through field by field, i.e. ensureSecurityContext

That's it. This is how the code that didn't ignore nil before has always worked, https://github.com/openshift/cluster-version-operator/blob/c4b0b4a98d5f02c47df121b1f9b923256a28cb24/lib/resourcemerge/core.go#L747

Reply:
I can add this comment which is what the `if` is doing:
`Check if we can simply set to required. This can be done if existing is not set or it is set but required is not set.`

Reply:
Added comments